### PR TITLE
add hosting of htdocs directory during private www build tests

### PR DESCRIPTION
**Pull request by charlesreid1**
Add hosting of htdocs directory in addition to hosting of log directory when running private-www build tests.

We also need to add simple username/password authentication with nginx on Uncle Archie (this is in the uncle-archie repo, so that is part of this pull request). Don't involve Heroku or Github logins, use nginx and keep it simple.

### CODE REVIEW

File: hooks/private_www_build_test.py [hunk 1]
```diff
@@ -178,26 +188,40 @@ def process_payload(payload, meta, config):
...
-                            context = params['task_name']
+                            context = params['build_task_name']
+            )
+        except GithubException as e:
+            logging.info("Github error: commit status failed to update.")
+
+        try:
+            commit_status = c.create_status(
+                            state = "success",
+                            target_url = status_url,
```
Comment (charlesreid1) [inline hooks/private_www_build_test.py]: `status_url_www`

File: hooks/private_www_build_test.py [hunk 2]
```diff
@@ -178,26 +188,40 @@ def process_payload(payload, meta, config):
...
         if build_msg == "":
-            build_msg = params['pass_msg']
+            build_msg = params['build_pass_msg']
+        if htdocs_msg == "":
+            htdocs_msg = params['htdocs_pass_msg']
 
         try:
             commit_status = c.create_status(
                             state = "success",
                             target_url = status_url,
```
Comment (charlesreid1) [inline hooks/private_www_build_test.py]: `status_url_log`

**Comment by charlesreid1** [comment]
~Oh yeah - we also need to fix this nonsense with "error" or "exception" in a commit message borking Uncle Archie. That's pretty annoying...~ (done)

~(Can just strip out the commit message verbatim, wherever it is, from the lower-cased string being checked for "error" or "exception")~ (done)

**Comment by charlesreid1** [comment]
~The existing URL schema for output docs is too complicated, make it easier for nginx to filter. Use URL schema below:~ (done)

mkdocs build output logs: <https://archie.nihdatacommons.us/output/logs/private_www_build_XXXXXXXX.log>

mkdocs build output static content: <https://archie.nihdatacommons.us/output/htdocs/private_www_build_XXXXXXXX/>

**Comment by charlesreid1** [comment]
We added a script to create an nginx username/password combination to the repo.

We merged branch `stuff` into branch `dcppc`, then merged branch `dcppc` into branch `add-htdocs-hosting`.

We checked out `add-htdocs-hosting` on beavo as user florence.

We created the .htaccess file and updated the nginx config file.

We copied the nginx config file into place:

```
sudo cp archie.conf /etc/nginx/sites-available/archie.nihdatacommons.conf
```

We activated the new nginx configuration:

```
sudo ln -fs /etc/nginx/sites-available/archie.nihdatacommons.conf /etc/nginx/sites-enabled/archie.nihdatacommons.conf
```

We restarted nginx:

```
sudo service nginx restart
```

We tested out creating a hello world page in `/www/archie.nihdatacommons.us/htdocs/output/htdocs/index.html`:

```
echo '<h2>hallo waerld</h2>' > /www/archie.nihdatacommons.us/htdocs/output/htdocs/index.html
```

We had a 403 permissions problem, caused by a problem with the script that created the .htaccess file (it did not put it in the right final location). Once that was fixed, we could access the hello world page in `/www/archie.nihdatacommons.us/htdocs/output/htdocs/index.html` successfully after authenticating with nginx.

In short - this is ready to go.

**Comment by charlesreid1** [comment]
It should be noted that currently this pull request only modifies the private-www build test to build a copy of the site. That's our initial target, so no problem. But we also want to do this for the four-year-plan and use-case-library.

**Comment by yammasnake** [comment]
Oh hey guys wutcha doin, can I help, I don't know what I'm doing but let me help

**Review by yammasnake — APPROVED** [review]
I have no idea what I'm doing

**Review by fp9695253 — APPROVED** [review]
I actually _do_ know what I'm doing

**Comment by fp9695253** [comment]
sheesh... you two...

**Comment by charlesreid1** [comment]
sheesh... you two...

**Comment by charlesreid1** [comment]
Note - we also need a script that will update `site_url` to the htdocs test URL

**Comment by charlesreid1** [comment]
^^^ that script is gonna need to be in the private-www build test... since we have to change the private-www mkdocs.yml before building

**Comment by charlesreid1** [comment]
See #39
